### DISCUSSION TITLE
DocC documentation hosting for Swift 5.7

### DISCUSSION THREAD

**Discussion by groue** [SwiftPackageIndex/SwiftPackageIndex-Server#2120] [Q&A] [answered]
Hello,

I wish SPI would host the DocC documentation for https://swiftpackageindex.com/groue/GRDB.swift, but the [SPI documentation](https://blog.swiftpackageindex.com/posts/auto-generating-auto-hosting-and-auto-updating-docc-documentation/) mentions:

> Ensure that your package builds successfully with Swift 5.6.

GRDB is 5.7+

Is there any hope?

Related discussion: https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/discussions/2010

**Comment by finestructure** [comment, marked as answer]
Oh, we need to update that. It should say 5.6+ and 5.7 is actually the default we use unless specified in `.spi.yml`.

**Reply by groue** [reply]
Thank you @finestructure :-)

OK, so I guess I missed something when I tried to host GRDB doc in SPI. I released v6.3.0 yesterday, with a [.spi.yml](https://github.com/groue/GRDB.swift/blob/v6.3.0/.spi.yml) file, but no documentation link shows up in the [SPI GRDB page](https://swiftpackageindex.com/groue/GRDB.swift) - even though the SPI bot has visited the repository. Is it because the `Package.swift` file is [unpractical to extend](https://github.com/groue/GRDB.swift/blob/v6.3.0/Package.swift#L39)?

(Yes, I read the fine print in the blog post _after_ pushing the changes 😅)

**Reply by finestructure** [reply]
That should work just fine and the `.spi.yml` file looks ok as well. I'll have a look what's going on!

**Reply by groue** [reply]
Thank you very much! 🙏

**Reply by finestructure** [reply]
Yeah, something is broken for the 5.7 doc build https://swiftpackageindex.com/builds/6547C124-73D5-4BE0-A1DA-2094E7C51CCD:

```
========================================
GenerateDocs
========================================
Generating docs at path:  spi-builder-workspace/.docs/groue/grdb.swift/v6.3.0
Repository:               groue/GRDB.swift
Swift version used:       5.7
    dependencies.append(.package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"))
Target:                   GRDB
error: Unknown subcommand or plugin name 'generate-documentation'
Us
age:
swift package <opti
ons
> <subcomma
nd>
  See 'package -help' for more information.
Error while generating docs: ShellOut encountered an error
Status code: 64
Message: "error: Unknown subcommand or plugin name 'generate-documentation'
Usage: swift package <options> <subcommand>
  See 'package -help' for more information."
Output: ""
Done.
```

Might be a machine issue, we were looking to upgrade it to Ventura's release version.

I'll convert this into an issue for us to look into. Thanks for the report!

**Reply by groue** [reply]
So we're helping each other 😉
